roles/mongodb-server/tasks/primary.yml

```diff
@@ -0,0 +1,18 @@
+---
+- name: initiate replica set
+  command: mongo --host {{ endpoints.mongodb }} --eval 'rs.initiate()'
+
+- name: wait for replica set init to complete
+  pause: seconds=20
+
+- name: rename primary replica member
+  command: mongo --host {{ endpoints.mongodb }} --eval 'cfg=rs.conf(); cfg.members[0].host="{{ primary_ip }}:{{ mongodb.port }}"; rs.reconfig(cfg)'
+
+- name: install pymongo for mongodb_user module
+  pip: name=pymongo state=present
+
+- name: create admin user
+  mongodb_user: database=admin name=admin password={{ mongodb.ceilometer_password }} roles='userAdminAnyDatabase' state=present
+
+- name: create ceilometer user
```

Comment:
these commands are all mongo commands for setting up the mongodb replica set, and the mongo/ceilometer users/databases. I don't think executing a mongo command should be done from a different role